### PR TITLE
Memoized bmi client

### PR DESCRIPTION
**Pull request by sverhoeven**
Wrapper around a Bmi instance which caches utility methods to prevent unneeded gRPC calls of utility methods during update loop.

### CODE REVIEW

File: grpc4bmi/bmi_memoized.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
+from bmi import Bmi
```
[the remark this answers is not in the document]
Comment (sverhoeven) [reply grpc4bmi/bmi_memoized.py]: The bmi-heat test dependency supplies bmi module. So I should not use that in the source code.

**Comment by Peter9192** [comment]
@sverhoeven installation problem seems to be fixed now. However, I have another question: the PR is against version 0.2.x, which is quite different from the latest master. Not sure if I understand which branch is used for what purpose.

**Comment by sverhoeven** [comment]
* In eWaterCycle the models have implemented a bmi interface described by https://pypi.org/project/basic-modeling-interface/ . The `0.2.x` branch uses this version of bmi.
* There is a newer version of bmi in https://pypi.org/project/bmipy/ . The `master` branch is compatible with bmipy.

It was the assumption that we would switch over to the bmipy version of bmi quickly. However the upgrade of the ewatercycle models has been delayed and we found several issues using 0.2.x version of grpc4bmi which caused the master and 0.2.x branches to diverge.

**Review by Peter9192 — APPROVED** [review]
Okay, thanks for the additional explanation. So I merged 0.2.x into my copy to test it and everything seems to work well. Have done a bit more testing, but I'm not entirely convinced that I see a significant speedup. The timing differences with/without memoizer in my tests are within the standard deviation of the test duration itself. However, one doesn't have to use it, and I guess we can do more thorough testing when we start using this in the comparison. So I'm okay with merging this.

**Comment by sverhoeven** [comment]
Thanks for reviewing. Will make a new release after merging so it available to users.

**Pull request merged**